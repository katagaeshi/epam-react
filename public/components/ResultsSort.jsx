import React from 'react';
import PropTypes from 'prop-types';
import RadioButtonGroup from './RadioButtonGroup';

const labelText = 'Sort by';

const ResultsSort = props => (
  <div>
    <label htmlFor="ResultsSort">{labelText}
      <RadioButtonGroup id="ResultsSort" {...props} />
    </label>
  </div>
);

ResultsSort.propTypes = {
  options: PropTypes.arrayOf(PropTypes.string),
  checked: PropTypes.string,
  idPrefix: PropTypes.string,
};

ResultsSort.defaultProps = {
  options: ['release date', 'rating'],
  checked: 'release date',
  idPrefix: 'ResultsSort-',
};

export default ResultsSort;
