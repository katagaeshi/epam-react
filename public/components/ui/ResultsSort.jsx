// @flow

import React from 'react';

import RadioButtonGroup from './RadioButtonGroup';

const labelText = 'Sort by';

type Props = {
  options: Array<string>,
  checked: string,
  idPrefix: string,
};

const ResultsSort = (props: Props) => (
  <div>
    <label htmlFor="ResultsSort">{labelText}
      <RadioButtonGroup idPrefix="ResultsSort" {...props} />
    </label>
  </div>
);

ResultsSort.defaultProps = {
  options: ['release date', 'rating'],
  checked: 'release date',
  idPrefix: 'ResultsSort-',
};

export default ResultsSort;
