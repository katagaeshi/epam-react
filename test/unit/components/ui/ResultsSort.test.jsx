import ResultsSort from 'public/components/ui/ResultsSort';

describe('ResultsSort', () => {
  it('Snapshot', () => {
    const tree = shallow(<ResultsSort
      onUpdate={() => {}}
    />);
    expect(tree).toMatchSnapshot();
  });
});
