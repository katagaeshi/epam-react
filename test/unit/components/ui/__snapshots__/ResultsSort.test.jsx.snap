// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsSort Snapshot 1`] = `
<div>
  <label
    htmlFor="ResultsSort"
  >
    Sort by
    <Jss(RadioButtonGroup)
      checked="release date"
      idPrefix="ResultsSort-"
      onUpdate={[Function]}
      options={
        Array [
          "release date",
          "rating",
        ]
      }
    />
  </label>
</div>
`;
